internal/server/singleprocess/service_url.go

```diff
@@ -11,6 +11,16 @@ import (
 )
 
 func (s *service) initURLGuestAccount(acceptURLTerms bool) error {
+	// Check if API Token already exists, if so, no reason to
+	// re-register and generate a new hostname
+	apiToken, err := s.state.ServerAPITokenGet()
+	if err != nil {
+		return err
+	} else if apiToken != "" {
+		// token already set, guest account already exists
```

Comment:
I think the success case has to set `s.urlConfig.APIToken` for this to work properly (see L52)